### ISSUE TITLE
vector format in defaults

### ISSUE THREAD

**Issue by jfree-man** [canmod/macpan2#188]
I think we need a better way to pass vectors as defaults.  Can we have an `mp_vector(y,labels)` function, that works similar to `mp_zero_vector` for a numeric vector `y` and optional character vector of labels `labels` if we want to give `y` names or not. This function would return the vector in the correct format for the simulator. 
A first start would be to better document the types of objects that can be passed as defaults, because I think it's likely other users may need to read in defaults from external files, and getting these vectors into the right format for the simulator isn't obvious.

The hypothetical `mp_vector` function has optional labels because:

- Sometimes we want the vector passed to the default list to have `names()`. Ex. if the specified model uses a state vector and individual state names in expressions. The object class of the vector, as well as attributes (names and dimnames) seem to determine whether the simulator will error or not. See [here](https://github.com/canmod/macpan2/blob/jen-dev/misc/experiments/defaults/simulator_defaults.R) for some experiments/motivation for this.
- Sometimes we don't need the vector passed to the default list to have `names()`. Ex. for the NFDS model, we read in an initial state vector from a csv file. This vector doesn't need labels because the state vector is ordered. I used `as.matrix()` and `unname()` to get the vector in an acceptable format for the simulator.

**Comment by stevencarlislewalker** [comment]
I'm thinking about this stuff now.  In the meantime, I feel like just passing ordinary numeric matrices either with or without names should be sufficient.  The standard R `setNames` function could be useful to get the functionality that you want.  Does this make sense?

**Comment by jfree-man** [comment]
Makes sense, not an urgent concern because I can work around this. I just tested using `setNames` on a matrix, and it doesn't seem to work...https://github.com/canmod/macpan2/blob/5bc71613cb6b7359db3c2ea001438abf0c2816e6/misc/experiments/defaults/simulator_defaults.R#L85

I'm thinking it's common for people to work with `data.frames` especially since this is the return value for `read.table`. So I'm wondering how we can communicate the format the table needs to be in. I'll think on this.

**Comment by stevencarlislewalker** [comment]
Thanks @jfree-man.  I'll think as well.

**Comment by bbolker** [comment]
`setNames` indeed doesn't work on matrices. Weirdly, matrices have "colnames", not "names".  We could certainly provide a helper function ...


```r
setColnames <- function(x, nm) {
    colnames(x) <- nm
    return(x)
}
``` 

Does that help?

**Comment by jfree-man** [comment]
It helps to know this information. So we can't use matrices with `colnames` (at the moment), because I think the simulator is expecting a `names` argument. We can pass unnamed matrices.

**Comment by stevencarlislewalker** [comment]
Thanks!  Would you mind pasting outputs from example results in 2?

**Comment by stevencarlislewalker** [comment]
Thanks Jen.  I didn't even think about 2iii, and 2iv.  But they are reasonable so thanks for adding them.
